### PR TITLE
Add types to transaction history

### PR DESCRIPTION
**Pull request by DavidTranDucVL**

### CODE REVIEW

File: app/frontend/components/pages/txHistory/transactionHistory.tsx [hunk 1]
```diff
@@ -114,13 +120,22 @@ const ExportCSV = ({transactionHistory, stakingHistory}) => {
...
-    received: 'Received',
-    sent: 'Sent',
-    rewardAwarded: 'Reward awarded',
+  const enum TransactionType {
+    received = 'Received',
+    sent = 'Sent',
+    rewardAwarded = 'Reward awarded',
+  }
+
+  type Entry = {
```
Comment (PeterBenc) [inline app/frontend/components/pages/txHistory/transactionHistory.tsx:129]: I would rename this at least to `TxEntry`

File: app/frontend/types.ts [hunk 2]
```diff
@@ -59,8 +60,21 @@ export type DerivationScheme = {
...
+export type Transaction = {
+  ctbId: HexString
+  ctbTimeIssued: number
+  ctbInputs: Array<Input>
+  ctbOutputs: Array<Output>
+  ctbInputSum: Coins
+  ctbOutputSum: Coins
+  fee: Lovelace
+  effect: Lovelace
+}
```
Comment (PeterBenc) [inline app/frontend/types.ts:77]: These are all defined in `explorer-types.ts`.

File: app/frontend/components/pages/txHistory/transactionHistory.tsx [hunk 3]
```diff
@@ -114,13 +120,22 @@ const ExportCSV = ({transactionHistory, stakingHistory}) => {
...
 
-  const transactionTypes = {
-    received: 'Received',
-    sent: 'Sent',
-    rewardAwarded: 'Reward awarded',
+  const enum TransactionType {
+    received = 'Received',
+    sent = 'Sent',
+    rewardAwarded = 'Reward awarded',
+  }
```
Comment (PeterBenc) [inline app/frontend/components/pages/txHistory/transactionHistory.tsx:127]: I would follow the convention and have enum entries with caps, so `RECEIVED = 'Received'`. As mentioned, some of these types might be defined in `explorer-types`. If not, I would rename this to `TxSummaryType` or something compliant with the other types.

**Review by PeterBenc — COMMENTED** [review]
Thnx, LGTM, you just probably missed the type declarations in `explorer-types.ts` and some names are IMO too general.

File: app/frontend/components/pages/txHistory/transactionHistory.tsx (hunk 1 above)
Comment (DavidTranDucVL) [reply app/frontend/components/pages/txHistory/transactionHistory.tsx]: Refactored

File: app/frontend/types.ts (hunk 2 above)
Comment (DavidTranDucVL) [reply app/frontend/types.ts]: I missed that, thanks.

File: app/frontend/components/pages/txHistory/transactionHistory.tsx (hunk 3 above)
Comment (DavidTranDucVL) [reply app/frontend/components/pages/txHistory/transactionHistory.tsx]: Refactored

**Review by PeterBenc — APPROVED** [review]

**Pull request merged**